test-graphql/client-test.js

```diff
@@ -0,0 +1,217 @@
+import assert from 'assert';
+import GraphQLJSClient from '@shopify/graphql-js-client';
+import Config from '../src-graphql/config';
+import Client from '../src-graphql/client';
+import types from '../types';
+import base64 from '../src-graphql/base64';
+import {singleProductFixture, multipleProductsFixture} from '../fixtures/product-fixture';
+import {fetchMock} from './fetch-mock-node';
+
+suite('client-test', () => {
+  const querySplitter = /[\s,]+/;
+
+  function tokens(queryString) {
+    return queryString.split(querySplitter).filter((token) => Boolean(token));
```

Comment:
Yeah even aside from the "query splitting", I do think it's useful to ensure we're generating valid queries but that should be done in `graphql-js-client` and not here I think. Any consumer of the client should be able to trust its queries.